File: backend/schemas/alert.py
from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel, Field


class AlertSeverityEnum(str, Enum):
    INFO = "info"
    WARNING = "warning"
    DANGER = "danger"
    CRITICAL = "critical"


class AlertResponse(BaseModel):
    id: str
    user_id: str
    account_id: str
    account_name: str = ""
    alert_type: str
    severity: str
    message: str
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AlertListResponse(BaseModel):
    alerts: list[AlertResponse]
    total: int


class AlertFilterParams(BaseModel):
    account_id: UUID | None = None
    severity: str | None = None
    alert_type: str | None = None
    status: str | None = None
    limit: int = Field(default=100, ge=1, le=500)

File: frontend/src/types/alert.ts
export type AlertSeverity = "info" | "warning" | "danger" | "critical";

export interface AlertNotification {
  id: string;
  user_id: string;
  account_id: string;
  account_name: string;
  alert_type: string;
  severity: AlertSeverity;
  message: string;
  status: string;
  created_at: string;
}

export interface AlertListResponse {
  alerts: AlertNotification[];
  total: number;
}

export interface AlertFilters {
  account_id?: string;
  severity?: string;
  alert_type?: string;
  status?: string;
  limit?: number;
}

export const ALERT_TYPE_LABELS: Record<string, string> = {
  daily_loss_70: "Daily Loss 70%",
  daily_loss_90: "Daily Loss 90%",
  daily_loss_violated: "Daily Loss Violated",
  drawdown_70: "Drawdown 70%",
  drawdown_90: "Drawdown 90%",
  drawdown_violated: "Drawdown Violated",
  profit_target_reached: "Profit Target Reached",
};

export const SEVERITY_OPTIONS = [
  { value: "", label: "All severities" },
  { value: "info", label: "Info" },
  { value: "warning", label: "Warning" },
  { value: "danger", label: "Danger" },
  { value: "critical", label: "Critical" },
];
